lib/chai/core/assertions.js

```diff
@@ -2145,4 +2145,28 @@ module.exports = function (chai, _) {
       , 'expected #{this} to not be frozen'
     );
   });
+
+  /**
+   * ### .finite
+   *
+   * Asserts that the target is a finite number. Unlike `.a('number')`, this will fail for `NaN` and `Infinity`.
+   *
+   *     expect(4).to.be.finite();
+   *     expect(NaN).to.not.be.finite();
+   *
+   * @name finite
+   * @namespace BDD
+   * @api public
+   */
+
+  Assertion.addMethod('finite', function(msg) {
+    if (msg) flag(this, 'message', msg);
+    var obj = flag(this, 'object');
+
+    this.assert(
+        typeof obj === "number" && isFinite(obj)
+      , 'expected #{this} to be a finite number'
+      , 'expected #{this} to not be a finite number'
+    );
+  });
```

Comment:
Yes I kind of agree; as much as I think we need to get rid of the property interface, we also need to stick to form more. If it takes no args, its a property.